### PR TITLE
Add types for react native dialog input

### PR DESCRIPTION
**Pull request by jeswinsimon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-native-dialog-input/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "name": "@types/react-native-dialog-input",
+    "version": "1.0.9999",
+    "projects": [
+        "https://github.com/joseestrella89/react-native-dialog-input#readme"
+    ],
+    "dependencies": {
+        "@types/react": "*",
+        "react-native": "*"
+    },
+    "devDependencies": {
```
Comment (codershiba) [inline types/react-native-dialog-input/package.json:12]:
```suggestion
    "devDependencies": {
              "@types/react": "*",
              "react-native": "*",
```

File: types/react-native-dialog-input/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (codershiba) [inline types/react-native-dialog-input/index.d.ts:1]:
```suggestion
import * as React from 'react';
```

**Comment by codershiba** [comment]
Revert commit 363f150

**Review by ghost — CHANGES_REQUESTED** [review]
Tests are missing

**Comment by jeswinsimon** [comment]
> Tests are missing

Added tests now.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-native-dialog-input/react-native-dialog-input-tests.tsx
1. types/react-native-dialog-input/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d32cf84363271212718ebbe4a7999071050dd321
</p>

**Review by peterblazejewicz — APPROVED** [review]
@jeswinsimon thanks!
LGTM!

**Comment by peterblazejewicz** [comment]
@codershiba thanks!

**Review by ghost — APPROVED** [review]

**Comment by jeswinsimon** [comment]
Ready to merge

**Pull request merged**